synapse/cores/sqlite.py

```diff
@@ -476,25 +476,28 @@ def _addRows(self, rows):
         with self.cursor() as c:
             c.executemany( self._q_addrows, rows )
 
-    def update(self, q, r, ret=False):
+    def update(self, q, r, ret=False, timeout=None):
         #print('UPDATE: %r %r' % (q,r))
         with self.cursor() as cur:
-            cur.execute(q,r)
+            self._execute(cur,q,r)
             if ret:
                 return cur.fetchall()
 
             return cur.rowcount
 
-    def select(self, q, r):
+    def select(self, q, r, timeout=None):
         #print('SELECT: %r %r' % (q,r))
         with self.cursor() as cur:
-            cur.execute(q,r)
+            self._execute(cur,q,r,timeout=timeout)
             return cur.fetchall()
 
-    def delete(self, q, r):
+    def delete(self, q, r, timeout=None):
```

Comment:
Leaving this kwarg to support _runPropQuery passing timeout to an arbitrary method.